CMakeLists.txt

```diff
@@ -9,7 +9,7 @@ include(LookUp-GreatCMakeCookOff)
 
 # Version and git hash id
 include(VersionAndGitRef)
-set_version(3.0.1)
+set_version(3.1.0)
```

Comment:
@mdavezac - do you know whether it's possible to append a `dev` to that version number? I would like to produce something like what we do in python:
`5.3.1dev556665` where `556665` is the hash number of the git where you build it. I've tried different things, but I wonder whether that type of `SONAME` is not allowed.

Would you suggest something better? cmake provides the githash, but I don't know how to get it back once the library is built. I've tried `readelf` but only gives me Major.Minor but not patch.

Reply:
If I remember well, this was mostly to reproduce the "sopt_about" executable that returned the sopt version number in the original c code. I'm not sure it's really useful.

In any case, if you look [here](https://github.com/UCL/GreatCMakeCookOff/blob/master/scripts/VersionAndGitRef.cmake), you'll find  the version can be anything you want. And there is a `get_gitref` function that will return the hash of the current commit.

To store it in the library, I guess the best thing would be add some static const globals with major, minor, patch and dev in the library. Then you could retrieve it by compiling a small program, or using readelf and friends?

Reply:
Thanks for the explanation!

I had understood the `get_gitref` function, but not how then the output was used or checked exactly by other software. The compiling bit is what we've done for some of the dependencies, but if there's not a defined standard...

Thanks again!